src/invision/components/movie-list/movie-list.js

```diff
@@ -1,90 +1,133 @@
+    const sortedMovieList = movieState.movieList.sort((movie1, movie2) => {
       return sortFn(movie1[sortType], movie2[sortType]);
     });
 
-    this.setState({ movieList: sortedMovieList });
+    updateMovieState({ ...movieState, movieList: sortedMovieList, sortValue });
+
+    function stringComparison(b, a) {
+      return a.localeCompare(b);
+    }
+
+    function numberComparison(b, a) {
+      if (a < b)
+        return -1;
+      if (a > b)
+        return 1;
+      return 0;
+    }
 
   }
-  stringComparison(b, a) {
-    return a.localeCompare(b);
+
+  const onModalClose = ({ visible, updatedMovie }) => {
+    if (updatedMovie) {
+      const index = movieState.movieList.findIndex(movie => movie.id === updatedMovie.id);
+      movieState.movieList[index] = { ...updatedMovie };
+    }
+    updateMovieState({ ...movieState, isEditMovieVisible: visible })
   }
 
-  numberComparison(b, a) {
-    if (a < b)
-      return -1;
-    if (a > b)
-      return 1;
-    return 0;
+  const onDeletePopupClose = (isDelete) => {
+    if (isDelete) {
+      // delete move code
+    } else {
+      updateMovieState({ ...movieState, isDeleteMovieVisible: false });
+    }
   }
 
+  const findMovie = useCallback((id) => {
+    return movieState.movieList.find(movie => movie.id === id);
+  })
 
-  render() {
-    return (
-      <section className="movie-list-section">
-        <div className="movie-list-ctnr">
-          <div className="header">
-            <div className="categories">
-              <div className="category">All</div>
-              <div className="category">DOCUMENTARY</div>
-              <div className="category">COMEDY</div>
-              <div className="category">HORROR</div>
-              <div className="category">CRIME</div>
-            </div>
-            <div className="sort">
-              SORT BY
-            <select onChange={this.sort} value={this.state.sortValue}>
-                <option value="RELEASE DATE">RELEASE DATE</option>
-                <option value="CATEGORY">CATEGORY</option>
-              </select>
-            </div>
+  const showEditPopup = (id) => {
+    const selectedMovie = findMovie(id);
+    updateMovieState({ ...movieState, selectedMovie, isEditMovieVisible: true })
+  }
+
+  const showDeletePopup = (id) => {
+    const selectedMovie = findMovie(id);
+    updateMovieState({ ...movieState, selectedMovie, isDeleteMovieVisible: true })
+  }
+
+  const showEditMoviePopup = () => {
+    if (movieState.isEditMovieVisible) {
+      return <MoviePopup visible="true" title="Edit Movie" movie={movieState.selectedMovie} onModalClose={onModalClose} />
+    }
+  }
+
+  const showDeleteMoviePopup = () => {
+    if (movieState.isDeleteMovieVisible) {
+      return <DeletePopup visible="true" title="Delete Movie" movie={movieState.selectedMovie} onDeletePopupClose={onDeletePopupClose} />
+    }
+  }
+
+  return (
+    <section className="movie-list-section">
+      { showEditMoviePopup()}
```

Comment:
Usually, to display something by condition, we do not create a function but use **ternary** operator - [documentation](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Conditional_Operator).
```
movieState.isEditMovieVisible 
          ? <MoviePopup visible="true" title="Edit Movie" movie={movieState.selectedMovie} onModalClose={onModalClose} />
          : null
```